### PR TITLE
- Improvement NonFactors#252, rows change not resetting current page …

### PR DESCRIPTION
**Pull request by jigneshsuvariya**
Improvement on bug fix - Instead of set pager to 1

Calculate new currentPage which can accommodate new pagesize.  

For example: 

If currentPage is 2 pageSize is 10 and totalRecodrs is 50.
Then when we change pageSize to 20 user should be stay on pageNumber 2.
But if we change pageSize to 50 new currentPage should be 1

### CODE REVIEW

File: src/Mvc.Grid/Views/Shared/MvcGrid/_Pager.cshtml [hunk 1]
```diff
@@ -1,77 +1,80 @@
 ﻿@using NonFactors.Mvc.Grid;
 @model IGridPager
 @{
+    Int32 totalRows = Model.TotalRows;
```
Comment (jigneshsuvariya) [inline src/Mvc.Grid/Views/Shared/MvcGrid/_Pager.cshtml]: Get total records

File: src/Mvc.Grid/Views/Shared/MvcGrid/_Pager.cshtml [hunk 2]
```diff
@@ -1,77 +1,80 @@
...
     Int32 totalPages = Model.TotalPages;
     Int32 currentPage = Model.CurrentPage;
     Int32 firstDisplayPage = Model.FirstDisplayPage;
 }
 
-    <div class="@($"mvc-grid-pager {Model.CssClasses}".Trim())" data-show-page-sizes="@Model.ShowPageSizes">
+<div class="@($"mvc-grid-pager {Model.CssClasses}".Trim())" data-show-page-sizes="@Model.ShowPageSizes">
     @if (totalPages > 0)
     {
+        <input type="hidden" class="mvc-grid-pager-totalRows" value="@totalRows" />
```
Comment (jigneshsuvariya) [inline src/Mvc.Grid/Views/Shared/MvcGrid/_Pager.cshtml]: Add value of total records to get into javascript

File: src/Mvc.Grid/js/mvc-grid/mvc-grid.js [hunk 3]
```diff
@@ -571,7 +572,10 @@ class MvcGridPager {
         }
 
         pager.rowsPerPage.addEventListener("change", () => {
-            pager.apply("1");
+            var totalPages = pager.totalRows / pager.rowsPerPage.value + ((pager.totalRows % pager.rowsPerPage.value) > 0 ? 1 : 0);
+            pager.currentPage = Math.min(totalPages, pager.currentPage);
+            pager.currentPage = Math.max(1, pager.currentPage);
+            pager.apply(pager.currentPage);
```
Comment (jigneshsuvariya) [inline src/Mvc.Grid/js/mvc-grid/mvc-grid.js]: Calculate new currentPage base on the totalRecords and New Page size

File: src/Mvc.Grid/js/mvc-grid/mvc-grid.ts [hunk 4]
```diff
@@ -662,7 +664,10 @@ export class MvcGridPager {
         }
 
         pager.rowsPerPage.addEventListener("change", () => {
-            pager.apply("1");
+            let totalPages: number = pager.totalRows / parseInt(pager.rowsPerPage.value) + ((pager.totalRows % parseInt(pager.rowsPerPage.value)) > 0 ? 1 : 0);
+            pager.currentPage = Math.min(totalPages, parseInt(pager.currentPage)).toString();
+            pager.currentPage = Math.max(1, parseInt(pager.currentPage)).toString();
+            pager.apply(pager.currentPage);
```
Comment (jigneshsuvariya) [inline src/Mvc.Grid/js/mvc-grid/mvc-grid.ts]: Calculate new currentPage base on the totalRecords and New Page size

**Comment by Muchiachio** [comment]
I think more logical would be to stay on the page which represents the currently viewed data.
As in 4th page of 10 row each paging should be the 2nd page of 20 row paging, of course even this would still run into edge cases of half page splits.

That's why current PR implementation is very similar to the "jump to first page", because user loses currently viewed data anyway. It also gives real numbers for pages in some scenarios.

**Comment by jigneshsuvariya** [comment]
What about take user to last page of new Page size? If the total number of pages after new Page Size is less than the currentPage ?

**Comment by Muchiachio** [comment]
It's already implemented in the grid, but only on the back end.
It didn't work for you because of manual data processing you have.

**Comment by jigneshsuvariya** [comment]
I can change my implementation in that way, If you agreed on that.

**Comment by Muchiachio** [comment]
Merged in 5c4fe78baf607bf56d8053a3b0e65b557986d723.

**Pull request closed without merging**